src/prefect/cli/_prompts.py

```diff
@@ -623,3 +642,178 @@ async def prompt_entrypoint(console: Console) -> str:
+            console,
+            prompt=(
+                f"Select from your existing {pretty_creds_block_type} credential blocks"
+            ),
+            columns=[
+                {
+                    "header": f"{pretty_storage_provider} Credentials Blocks",
+                    "key": "name",
+                }
+            ],
+            data=[{"name": block.name} for block in existing_credentials_blocks],
+            opt_out_message="Create a new credentials block",
+        )
+
+        if selected_credentials_block and (
+            selected_block := selected_credentials_block.get("name")
+        ):
+            return f"{{{{ prefect.blocks.{creds_block_type_slug}.{selected_block} }}}}"
+
+    credentials_block_type = await client.read_block_type_by_slug(creds_block_type_slug)
+
+    credentials_block_schemas = (
+        await client._client.post(
+            "/block_schemas/filter",
+            json={
+                "block_schemas": {
+                    "block_type_id": {"any_": [str(credentials_block_type.id)]}
+                },
+                "limit": 1,
+            },
+        )
+    ).json()
+
+    credentials_block_schema = parse_obj_as(BlockSchema, credentials_block_schemas[0])
+
+    console.print(
+        f"\nProvide details on your new {pretty_storage_provider} credentials:"
+    )
+
+    hydrated_fields = {
+        field_name: prompt(f"{field_name} [yellow]({props.get('type')})[/]")
+        for field_name, props in credentials_block_schema.fields.get(
+            "properties"
+        ).items()
+        if field_name in REQUIRED_FIELDS_FOR_CREDS_BLOCK[creds_block_type_slug]
+    }
+
+    console.print(f"[blue]\n{pretty_storage_provider} credentials specified![/]\n")
+
+    credentials_block_name = prompt(
+        "Give a name to your new credentials block",
+        default=f"{storage_provider_slug}-storage-credentials",
+    )
+
+    try:
+        new_block_document = await client.create_block_document(
+            block_document=BlockDocumentCreate(
+                name=credentials_block_name,
+                data=hydrated_fields,
+                block_schema_id=credentials_block_schema.id,
+                block_type_id=credentials_block_type.id,
+            )
+        )
+    except ObjectAlreadyExists:
+        credentials_block_name = prompt(
+            (
+                f"A {pretty_creds_block_type!r} block named"
+                f" {credentials_block_name!r} already exists. Please choose another"
+                " name"
+            ),
+            default=f"{storage_provider_slug}-storage-credentials",
+        )
+
+        new_block_document = await client.create_block_document(
+            block_document=BlockDocumentCreate(
+                name=credentials_block_name,
+                data=hydrated_fields,
+                block_schema_id=credentials_block_schema.id,
+                block_type_id=credentials_block_type.id,
+            )
+        )
```

Comment:
If you build a custom prompt, this part can go away.

Reply:
because i want to return the block document out of the prompt loop, it seemed slightly more complex to use a custom prompt here - pushing up the alternative now, lemme know what you think

Reply:
Yup, that should work.